include/bsp/iotc_test_echoserver.h

```diff
@@ -0,0 +1,34 @@
+#ifndef __IOTC_TEST_ECHOSERVER_H__
+#define __IOTC_TEST_ECHOSERVER_H__
+
+#include <netdb.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+const uint16_t kTimeoutSeconds = 1;
+const uint16_t kBufferSize = 1024;
+class EchoTestServer {
+public:
+  EchoTestServer(uint16_t sockType, uint16_t port)
+      : sock_type_(sockType), test_port_(port) {}
+
+  int recv_len_;
```

Comment:
Move the members that should be private into `private:` section. For example `recv_len_`, `recv_buf_`, ..

Some methods should also be private. `runTcpServer()` and `runUdpServer()` should not be exposed to the user, as it allows to do stupid stuff (ex: call `runTcpServer()` when they created the instance with `sockType` = `SOCK_DGRAM`).